controllers/data.controller.js

```diff
@@ -99,68 +105,66 @@ const handleClassicQuestions = async (
   isAnswerCorrect,
   questionsAnsweredNb,
 ) => {
-  console.log("handleClassicQuestions");
-  try {
-    const updatedScore = isAnswerCorrect
-      ? currentScore + 80
-      : currentScore - 50;
-    const tierIndex = TIER_LIST.indexOf(currentTier);
-    // demote after 2 wrong answers in a row
-    if (updatedScore < -50) {
-      const newScore = 100 + updatedScore; // score is negative so we add it to 100 (ex : 100 + (-50) = 50)
-      const newTier = TIER_LIST[tierIndex - 1];
-      const response = await User.findOneAndUpdate(
-        { _id: uid },
-        {
-          $set: {
-            score: newScore,
-            tier: newTier,
-            questionsAnsweredNb: questionsAnsweredNb + 1,
-          },
+  const updatedScore = isAnswerCorrect ? currentScore + 80 : currentScore - 50;
+  const tierIndex = TIER_LIST.indexOf(currentTier);
+  let response = null;
+  // demote after 2 wrong answers in a row
+  if (updatedScore < -50) {
+    const newScore = 100 + updatedScore; // score is negative so we add it to 100 (ex : 100 + (-50) = 50)
+    const newTier = TIER_LIST[tierIndex - 1];
+    response = await User.findOneAndUpdate(
```

Comment:
I think you do it outside of "if else" function, so you dont need to write this three times. just declare newScore and newTier globally and you can change it in "if else" function and then use it outside.

Reply:
Here the calcul is slightly different for each condition so I can't just define it the same way for all conditions

Reply:
> I think you do it outside of "if else" function, so you dont need to write this three times. just declare newScore and newTier globally and you can change it in "if else" function and then use it outside.

Here the calcul is slightly different for each condition so I can't just define it the same way for all conditions

Reply:
@axelgcd you dont need to define in the same way. you can use "if else" for newScore and newTier, and outside of it apply to  response = await User.findOneAndUpdate